server/repository/src/test_db/sqlite.rs

```diff
@@ -1,59 +1,157 @@
-    DBBackendConnection, StorageConnection, StorageConnectionManager,
+    mock::{all_mock_data, insert_all_mock_data, MockDataCollection, MockDataInserts},
+    DBBackendConnection, StorageConnectionManager,
 };
 
+const TEST_OUTPUT_DIR: &'static str = "test_output";
+const TEMPLATE_MARKER_FILE: &'static str = "___template_needs_update.marker";
+
 pub fn get_test_db_settings(db_name: &str) -> DatabaseSettings {
     DatabaseSettings {
         username: "postgres".to_string(),
         password: "password".to_string(),
         port: 5432,
         host: "localhost".to_string(),
         // put DB test files into a test directory (also works for in-memory)
-        database_name: format!("test_output/{}.sqlite", db_name),
+        database_name: format!("{}/{}.sqlite", TEST_OUTPUT_DIR, db_name),
         init_sql: None,
     }
 }
 
 pub async fn setup(db_settings: &DatabaseSettings) -> StorageConnectionManager {
-    setup_with_version(db_settings, None).await
+    setup_with_version(db_settings, None, MockDataInserts::none())
+        .await
+        .0
 }
 
+static TEMPLATE_LOCK: Mutex<()> = Mutex::new(());
+
 pub(crate) async fn setup_with_version(
     db_settings: &DatabaseSettings,
     version: Option<Version>,
-) -> StorageConnectionManager {
+    inserts: MockDataInserts,
+) -> (StorageConnectionManager, MockDataCollection) {
     let db_path = db_settings.connection_string();
 
-    // If not in-memory mode clean up and create test directory
-    // (in in-memory mode the db_path starts with "file:")
-    if !db_path.starts_with("file:") {
+    let (connection_manager, collection) = if db_path.starts_with("file:") {
+        // memory mode
+        let connection_manager = create_db(&db_settings, version.clone());
+        let connection = connection_manager.connection().unwrap();
+        let collection = insert_all_mock_data(&connection, inserts).await;
+        (connection_manager, collection)
+    } else {
+        // cache db template
+        let cache_all_mock_data = inserts == MockDataInserts::all();
+        let template_name = if cache_all_mock_data {
+            format!(
+                "___template_{}_full_mock",
+                version.as_ref().unwrap_or(&Version::from_package_json())
+            )
+        } else {
+            format!(
+                "___template_{}",
+                version.as_ref().unwrap_or(&Version::from_package_json())
+            )
+        };
+
+        let guard = TEMPLATE_LOCK.lock().unwrap();
+        // if marker exists, DB needs to be recreated -> delete all template files
+        let marker_path =
+            Path::new(&format!("{}/{}", TEST_OUTPUT_DIR, TEMPLATE_MARKER_FILE)).to_path_buf();
+        if marker_path.exists() {
+            // remove all DB templates
+            for entry in fs::read_dir(TEST_OUTPUT_DIR).unwrap() {
+                let entry = entry.unwrap();
+                if entry.file_name().to_string_lossy() == TEMPLATE_MARKER_FILE {
```

Comment:
Yes good point, was about to change but couldn't come up with a better name and then thought it might be good to explicitly ensure we don't delete the file before cleaning up the templates...